app/assets/javascripts/reports/save_pdf_to_inventory.js.erb

```diff
@@ -0,0 +1,287 @@
+  }
+
+  function appendSearchResultsForColumns(data) {
+    var items = [];
+    if(data.hasOwnProperty('results')){
+      $('#selectInventoryColumn').parent().find('button').removeAttr('disabled');
+      $.each(data.results, function(index, el) {
+        items.push(
+          {
+            value: el.id,
+            text: el.name,
+            disabled: false
+          }
+        )
+      });
+    } else {
+      $('#selectInventoryColumn').parent().find('button').attr('disabled', true);
+      clearErrors();
+      $('#save-PDF-to-inventory-column-warnings').append(
+        '<div class="save-PDF-to-inventory-alerts"><strong class="danger">' +
+        data.no_items + '</strong></div>'
+      )
+    }
+    return items;
+  }
+
+  function submitButtonEnableToggle(status) {
+    var button = $('#savePDFtoInventorySubmit');
+    if(status) {
+      button.removeAttr('disabled');
+    } else {
+      button.attr('disabled', true);
+    }
+  }
+
+  function deselect(element) {
+    if(element) {
+      element.selectpicker('val', null)
+      element.selectpicker('render');
+      element.attr('disabled', true);
+    }
+  }
+
+  // triggers first request and cleans the dropdown selectpicker
+  function clearDropdownResultsCallback(element) {
+    element.parent().find('button').on('click', function(el) {
+      $(this).parent().find('input').trigger('keyup');
+    });
+  }
+
+  function initInventoryItemSelectPicker() {
+    ITEM_PICKER =
+      $('#selectInventoryItem')
+        .removeAttr('disabled')
+        .selectpicker({liveSearch: true})
+        .ajaxSelectPicker({
+          ajax: {
+            url: '<%= Rails.application.routes.url_helpers.available_rows_path %>',
+            type: 'POST',
+            dataType: 'json',
+            data: function () {
+              return {
+                q: '{{{q}}}',
+                repository_id: SELECTED_IDS.repository_id,
+                repository_column_id: SELECTED_IDS.respository_column_id
+              };
+            }
+          },
+          locale: {
+            emptyTitle: '<%= I18n.t('projects.reports.new.save_PDF_to_inventory_modal.nothing_selected') %>'
+          },
+          preprocessData: appendSearchResultsForItems,
+          emptyRequest: true,
+          clearOnEmpty: true,
+          cache: false,
+          preserveSelected: false
+        }).on('change.bs.select', function(el) {
+          var value = el.target.value;
+          toggleHasFileErrorMessage(value);
+          submitButtonEnableToggle(true);
+          SELECTED_IDS.repository_item_id = value;
+        });
+    clearDropdownResultsCallback(ITEM_PICKER);
+  }
+
+  function initInventoryColumnSelectPicker() {
+    COLUMN_PICKER =
+      $('#selectInventoryColumn')
+        .removeAttr('disabled')
+        .selectpicker({liveSearch: true})
+        .ajaxSelectPicker({
+          ajax: {
+            url: '<%= Rails.application.routes.url_helpers.file_columns_path %>',
+            type: 'POST',
```

Comment:
https://github.com/truckingsim/Ajax-Bootstrap-Select#optionsajax

Reply:
Well, this is just an example code snippet, it doesn't dictate you to use `POST`. RESTful specification would dictate use of `GET` for this operation. On the other hand, browsers can cache `GET` requests, so in this case, `POST` would make more sense to prevent that situation from happening. So we can keep the `POST`.